Resources/Storage/GFAL2_StorageBase.py

```diff
@@ -965,19 +957,17 @@ def __getChecksum( self, path, checksumType = None ):
     res = self.__isSingleFile( path )
 
     if not res['OK']:
-      errStr = 'GFAL2_StorageBase.__getChecksum: provided path is not a file'
-      self.log.error( errStr, path )
-      return S_ERROR( errStr )
-    else:
-      try:
-        self.log.debug( "GFAL2_StorageBase.__getChecksum: using %s checksum" % checksumType )
-        fileChecksum = self.gfal2.checksum( path, checksumType )
-        return S_OK( fileChecksum )
+      return res
+
+    try:
+      self.log.debug( "GFAL2_StorageBase.__getChecksum: using %s checksum" % checksumType )
+      fileChecksum = self.gfal2.checksum( path, checksumType )
+      return S_OK( fileChecksum )
 
-      except gfal2.GError, e:
-        errStr = 'GFAL2_StorageBase.__getChecksum: failed to calculate checksum.'
-        self.log.error( errStr, e.message )
-        return S_ERROR( e.message )
+    except gfal2.GError as e:
+      errStr = 'GFAL2_StorageBase.__getChecksum: failed to calculate checksum.'
+      self.log.error( errStr, repr( e ) )
+      return S_ERROR( e.code, repr( e ) )
```

Comment:
Previously you also used `return S_ERROR( e.code, e.message )` and here `return S_ERROR( e.code, repr( e ) )`.

What's better?